medium-tests/src/test/java/mediumtest/synchronization/ConnectionSyncMediumTests.java

```diff
@@ -137,6 +139,41 @@ void it_should_sync_when_credentials_are_updated() {
       "Synchronizing project branches for project 'projectKey'"));
   }
 
+  @ParameterizedTest
+  @ValueSource(ints = {401, 403})
+  void it_should_notify_client_if_invalid_token(Integer status) {
+    var client = newFakeClient()
+      .withCredentials(CONNECTION_ID, "user", "pw")
+      .build();
+    when(client.getClientLiveDescription()).thenReturn(this.getClass().getName());
+
+    var introductionDate = Instant.now().truncatedTo(ChronoUnit.MILLIS);
+    var server = newSonarQubeServer()
+      .withProject("projectKey",
+        project -> project.withBranch("main",
+          branch -> branch.withTaintIssue("issueKey", "rule:key", "message", "author", "file/path", "OPEN", null, introductionDate, new TextRange(1, 2, 3, 4),
+            RuleType.VULNERABILITY)))
+      .start();
+
+    server.getMockServer().stubFor(get("/api/system/status").willReturn(aResponse().withStatus(status)));
```

Comment:
I think it would make sense to improve the server fixture instead of accessing the internals. It might be useful for other cases in the future, and it hides how we implement the fixture